heft-plugins/heft-storybook-plugin/src/StorybookPlugin.ts

```diff
@@ -0,0 +1,237 @@
+      throw new Error(
+        `The ${TASK_NAME} task cannot start because the "storykitPackageName"` +
+          ` plugin option is not a valid package name: ` +
+          parseResult.error
+      );
+    }
+    this._storykitPackageName = options.storykitPackageName;
+
+    if (!options.startupModulePath) {
+      throw new Error(
+        `The ${TASK_NAME} task cannot start because the "startupModulePath"` +
+          ` plugin option was not specified`
+      );
+    }
+    this._startupModulePath = options.startupModulePath;
+
+    heftSession.hooks.build.tap(PLUGIN_NAME, (build: IBuildStageContext) => {
+      // TODO: Expose an API for custom CLI parameters similar to HeftSession.registerAction()
+      if (process.argv.indexOf('--storybook') < 0) {
+        this._logger.terminal.writeVerboseLine(
+          'The command line does not include "--storybook", so bundling will proceed without Storybook'
+        );
+        return;
+      }
+
+      this._logger.terminal.writeVerboseLine(
+        'The command line includes "--storybook", redirecting Webpack to Storybook'
+      );
+
+      build.hooks.preCompile.tap(PLUGIN_NAME, (preCompile: IPreCompileSubstage) => {
+        preCompile.hooks.run.tap(PLUGIN_NAME, () => {
+          this._onPreCompile(heftSession, heftConfiguration);
+        });
+      });
+
+      build.hooks.bundle.tap(PLUGIN_NAME, (bundle: IBundleSubstage) => {
+        bundle.hooks.configureWebpack.tap(
+          { name: PLUGIN_NAME, stage: Number.MAX_SAFE_INTEGER },
+          (webpackConfiguration: unknown) => {
+            // Discard Webpack's configuration to prevent Webpack from running
+            return null;
+          }
+        );
+
+        bundle.hooks.run.tapPromise(PLUGIN_NAME, async () => {
+          await this._onBundleRunAsync(heftSession, heftConfiguration);
+        });
+      });
+    });
+  }
+
+  private _onPreCompile(heftSession: HeftSession, heftConfiguration: HeftConfiguration): void {
+    this._logger.terminal.writeVerboseLine(`Probing for "${this._storykitPackageName}"`);
+
+    // Example: "/path/to/my-project/node_modules/my-storykit"
+    let storykitFolder: string;
+    try {
+      storykitFolder = Import.resolvePackage({
+        packageName: this._storykitPackageName,
+        baseFolderPath: heftConfiguration.buildFolder
+      });
+    } catch (ex) {
+      throw new Error(`The ${TASK_NAME} task cannot start: ` + ex.message);
+    }
+
+    this._logger.terminal.writeVerboseLine(`Found "${this._storykitPackageName}" in ` + storykitFolder);
+
+    // Example: "/path/to/my-project/node_modules/my-storykit/node_modules"
+    const storykitModuleFolder: string = path.join(storykitFolder, 'node_modules');
+    if (!FileSystem.exists(storykitModuleFolder)) {
+      throw new Error(
+        `The ${TASK_NAME} task cannot start because the storykit module folder does not exist:\n` +
+          storykitModuleFolder +
+          '\nDid you forget to install it?'
+      );
```

Comment:
```suggestion
      this._logger.emitError(new Error((
        `The ${TASK_NAME} task cannot start because the storykit module folder does not exist:\n` +
          storykitModuleFolder +
          '\nDid you forget to install it?'
      );
      return;
```

Reply:
(See [my other note](https://github.com/microsoft/rushstack/pull/2885#discussion_r707821227))